### PR TITLE
[cloud_functions] web plugin implementation

### PR DESCRIPTION
**Pull request by sbeitzel**
## Description

This PR delivers the web implementation of the cloud_functions plugin. Note that although README.md refers to the example in cloud_functions, that can't actually be updated until this plugin gets published once (otherwise version resolving will fail).

## Related Issues

Resolves issue [45299](https://github.com/flutter/flutter/issues/45299)

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by sbeitzel** [comment]
It looks like the failure in [build-ipas+drive-examples CHANNEL:master PLUGIN_SHARDING:--shardIndex 1 --shardCount 4](https://github.com/FirebaseExtended/flutterfire/pull/1890/checks?check_run_id=413299317) is not because of the plugin code but because that build check sets up for iOS 13.3 and then tries to run a simulator with iOS 13.2, which hasn't been installed. Who's in charge of configuring the check scripts?

File: packages/cloud_functions/cloud_functions_web/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# cloud_functions_web
+
+The web implementation of [`cloud_functions`][1].
+
+## Usage
+
+### Import the package
+
+This package is the endorsed implementation of `cloud_functions` for the web platform since version `0.0.1`, so it gets automatically added to your dependencies by depending on `cloud_functions: ^0.0.1`.
```
Comment (hterkelsen) [inline packages/cloud_functions/cloud_functions_web/README.md]: Put a TODO here to update this with the correct version number as soon as this package actually becomes an endorsed implementation of `package:cloud_functions`

File: packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:cloud_functions_platform_interface/cloud_functions_platform_interface.dart';
+import 'package:firebase/firebase.dart' as firebase;
+import 'package:flutter/foundation.dart';
```
Comment (hterkelsen) [inline packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart]: unused import?

File: packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+
+/// Web implementation of [CloudFunctionsPlatform]
```
Comment (hterkelsen) [inline packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart]: punctuation

File: packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart (hunk 2 above)
Comment (sbeitzel) [reply packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart]: foundation.dart is the source of the @required annotation in the callCloudFunction method signature

File: packages/cloud_functions/cloud_functions_web/README.md (hunk 1 above)
Comment (sbeitzel) [reply packages/cloud_functions/cloud_functions_web/README.md]: Oh, good catch. Sorry, will do that right now.

**Comment by flowhorn** [comment]
Is there support for other regions? Because in the used firebase package there is no support right now and the region parameter isn't used I think. Or am I wrong?

**Comment by sbeitzel** [comment]
This implementation delegates to the [firebase](https://pub.dev/packages/firebase) plugin and you're right, the region parameter is silently ignored because there's not an obvious way to pass it down.

**Comment by hterkelsen** [comment]
You can set the region by doing `firebase.app().functions('region-name')`: https://github.com/FirebaseExtended/firebase-dart/blob/master/lib/src/app.dart#L58

File: packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  @override
+  Future<dynamic> callCloudFunction({
+    @required String appName,
+    @required String functionName,
+    String region,
+    String origin,
+    Duration timeout,
+    dynamic parameters,
+  }) {
+    firebase.Functions functions = firebase.functions(firebase.app(appName));
```
Comment (hterkelsen) [inline packages/cloud_functions/cloud_functions_web/lib/cloud_functions_web.dart]: Set the region by instantiating `firebase.Functions` another way: https://github.com/FirebaseExtended/firebase-dart/blob/master/lib/src/app.dart#L58

**Pull request merged**